ddm_predictor.py

```diff
@@ -57,14 +58,30 @@ def __init__(
         self.state_keys = states
         self.action_keys = actions
         self.diff_state = diff_state
+        self.initial_states = initial_states
         # TODO: Add logging
 
         logger.info(f"DDM features: {self.features}")
         logger.info(f"DDM outputs: {self.labels}")
 
     def episode_start(self, config: Dict[str, Any] = None):
-
-        initial_state = {k: random.random() for k in self.state_keys}
+        """Initial DDM with initial states. This could include initializations of configs
+        as well as initial values for actions
+
+        Parameters
+        ----------
+        config : Dict[str, Any], optional
+            episode initializations, by default None
+        """
+
+        # initialize states based on simulator.yaml
+        initial_state = self.initial_states
+        # define initial state from config if available (e.g. when brain training)
+        # skip if config missing
+        if config:
+            initial_state.update(
+                (k, config[k]) for k in initial_state.keys() & config.keys()
```

Comment:
This would require the config in inkling and initial_state config in the yaml file to have same key values (atleast some of them). So, while writing configs in yaml file the user has to make sure they use same name as in inkling. In other cases (or previous case), that was not required. Probably commenting should be fine.